linux/deploy_vm/rebuild_unattend_install_iso.yml

```diff
@@ -0,0 +1,122 @@
+
+- name: "Set fact of the temp dir to mount OS ISO image"
+  set_fact:
+    tmp_iso_mount_dir: "{{ tmp_path }}"
+
+# Mount original ISO image at local
+- include_tasks: ../../common/local_mount.yml
+  vars:
+    mount_path: "{{ tmp_iso_mount_dir }}"
+    mount_src: "{{ src_iso_file_path }}"
+    mount_opts: "ro,noauto"
+    mount_fstype: "iso9660"
+
+# Copy files and directories in original ISO to a new folder
+# And add unattended install file
+- block:
+    - name: "Create a temporary directory to rebuild unattended install ISO image"
+      include_tasks: ../../common/create_temp_file_dir.yml
+      vars:
+        tmp_dir: "{{ unattend_iso_cache }}"
+        tmp_state: "directory"
+        tmp_prefix: "rebuild_iso_"
+
+    - name: "Set facts for rebuilding unattend install ISO image"
+      set_fact:
+        rebuilt_unattend_iso_dir: "{{ tmp_path }}"
+
+    - name: "Set fact of unattend install config file name"
+      set_fact:
+        unattend_install_file_name: "{{ new_unattend_install_conf | basename }}"
+
+    - name: "Set fact of boot menu config file"
+      set_fact:
+        boot_menu_cfg_file_paths:
+          - "{{ rebuilt_unattend_iso_dir }}/isolinux/menu.cfg"
+          - "{{ rebuilt_unattend_iso_dir }}/boot/grub2/grub.cfg"
+        isolinux_cfg_file_path: "{{ rebuilt_unattend_iso_dir }}/isolinux/isolinux.cfg"
+
+    - block:
+        # Ansible copy module would fail while copying read-only files
+        - name: "Copy ISO content from {{ tmp_iso_mount_dir }} to {{ rebuilt_unattend_iso_dir }}"
+          shell: "cp -rf {{ tmp_iso_mount_dir }}/* {{ rebuilt_unattend_iso_dir }}"
+
+        - name: "Copy unattend install config file to {{ rebuilt_unattend_iso_dir }}"
+          copy:
+            src: "{{ new_unattend_install_conf }}"
+            dest: "{{ rebuilt_unattend_iso_dir }}/isolinux"
+            force: yes
+            mode: 0644
+
+        - name: "Update timeout for autoboot"
+          replace:
+            path: "{{ isolinux_cfg_file_path }}"
+            regexp: "timeout 0"
+            replace: "timeout 1"
+
+        - name: "Update boot menu with kickstart"
+          replace:
+            path: "{{ item }}"
+            regexp: '(.*)(root=[^ ]+)(.*)'
+            replace: "\\1\\2 ks=cdrom:/isolinux/{{ unattend_install_file_name }} \\3"
+          with_items: "{{ boot_menu_cfg_file_paths }}"
+
+        - name: "Set facts for rebuilding ISO options"
+          set_fact:
+            unattend_iso_label: "-V {{ src_iso_file_name | splitext | first }}"
+            boot_info_opts: "-R -l -L -D -b isolinux/isolinux.bin -c isolinux/boot.cat -no-emul-boot -boot-load-size 4 -boot-info-table -eltorito-alt-boot --eltorito-boot boot/grub2/efiboot.img -no-emul-boot"
+
+        # Create the unattended install ISO image
+        - name: "Rebuild ISO image with unattend install config file"
+          shell: "xorriso -as mkisofs {{ boot_info_opts }} {{ unattend_iso_label }} . > {{ rebuilt_unattend_iso_path }}"
```

Comment:
Does this "xorriso" package always available and need to be added in requirements.txt?

Reply:
Yes, it is required. I just updated requirements.txt. Once the PR merged, we need to update the docker image as well.